### PR TITLE
add application tests, fix EventSystem clean up bug

### PR DESCRIPTION
**Pull request by achamas-playco**
### <samp>🤖 Generated by Copilot at 470b1f1</samp>

This pull request adds support for OffscreenCanvas rendering in pixijs, fixes a memory leak issue in the GpuTextureSystem, and improves the test coverage and utility for the Application class. It modifies the `EventSystem` and `GpuTextureSystem` classes to handle edge cases and errors, and adds new test files and functions for the `Application` class and its options.

### CODE REVIEW

**Review by Zyie — APPROVED** [review]

File: tests/app/Application.tests.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+describe('Application', () =>
+{
+    const setup = async (options?: Partial<ApplicationOptions>) =>
+    {
+        const app = new Application();
+
+        await app.init(options);
+
+        return app;
+    };
```
Comment (Zyie) [inline tests/app/Application.tests.ts:16]: Will other tests also need to create an app?
If so we could move this to a utils file, if not feel free to merge

Comment (achamas-playco) [reply tests/app/Application.tests.ts]: Added `getApp` util function to help with other tests
